### PR TITLE
feat: Implemented LIC6 #20

### PR DESCRIPTION
**Pull request by andersblomqvist**
New feature: LIC6 calculation

Added relevant tests which are all passing✅

### CODE REVIEW

File: src/main/CMV.java [hunk 1]
```diff
@@ -94,7 +94,54 @@ public boolean lic5_calculate() {
...
+        int N_PTS = Parameters.N_PTS;
+        double DIST = Parameters.DIST;
+        if(N_PTS < 3 || N_PTS > datapoints.length || DIST <= 0)
+            return false;
+        
+        // Go through each set of N_PTS consecutive points
+        for (int point = 0; point < datapoints.length; point++) {
+            // First and last point within the set of consecutive points
+            int p1 = point;
+            int p2 = (point + N_PTS - 1) % datapoints.length;
```
Comment (Darpos) [inline src/main/CMV.java:107]: unsure wheter p2 > datapoints.length should loop back to the start. Don't know if that is really considered _consecutive_.

File: src/main/CMV.java [hunk 2]
```diff
@@ -94,7 +94,54 @@ public boolean lic5_calculate() {
         return false;
     }
 
-    private Boolean lic6_calculate() {
+    public Boolean lic6_calculate() {
```
Comment (Darpos) [inline src/main/CMV.java:97]: I am unsure wheter this is the correct solution to the assignment. From reading the description it sounds like a projection has to be made from the line onto each point (or the other way around) in order to get a appropiate distance.

**Review by Darpos — COMMENTED** [review]
Unsure if this is correct, I am thinking the solution has to utilize this: https://en.wikibooks.org/wiki/Linear_Algebra/Orthogonal_Projection_Onto_a_Line. Requesting ohers to review and see how to interpret the assignment.

File: src/main/CMV.java (hunk 2 above)
Comment (andersblomqvist) [reply src/main/CMV.java:97]: This is what I would argue that I'm doing.

1. The first for loop goes through each consecutive number of nodes and picks out the first and last points.
2. The second for loop goes through each point in between first and last, checking if the distance to the line is greater than `DIST`.

The distance equation is from: [Line defined by two points](https://en.wikipedia.org/wiki/Distance_from_a_point_to_a_line#Line_defined_by_two_points)

File: src/main/CMV.java (hunk 1 above)
Comment (andersblomqvist) [reply src/main/CMV.java:107]: I agree with your view on consecutive order. Will change that.

**Review by coffecup25 — CHANGES_REQUESTED** [review]
The testing looks good but the lic6_calculate() should be a private function following pull request #54

**Review by Darpos — APPROVED** [review]
After reading the comment addressing my comment I 100% that this is a correct solution! Good job.

**Review by coffecup25 — APPROVED** [review]

**Comment by andersblomqvist** [comment]
Closes #20

**Pull request merged**